### ISSUE TITLE
It's not generating my token correctly

### ISSUE THREAD

**Issue by krainboltgreene** [ingydotnet/git-hub#163]
```
▶ git hub token-new

Can't determine your Personal API Access Token, which is required by this
command. Usually you put your token into the ~/.git-hub/config file, like this:

  git hub config api-token <your-personal-api-access-token>

You can also specify the token using the --token= commandline option, or via
the GIT_HUB_API_TOKEN environment variable.

To list your API tokens:

  git hub tokens

To create a new api token:

  git hub token-new "my git-hub token"
  # You should probably add at least the 'repo' and 'user' scopes:
  git hub scope-add <token-id> repo user

You can also just run setup command, and it will guide you:

  git hub setup

Would you like to run it now?

Press <ENTER> to continue, or <CTL>-C to exit.

                        The 'git hub' Setup Tool

This dialog will walk you through the steps involved in creating or updating
the configuration needed by the 'git hub' command. It should only take a
minute.

Your config will be stored here: '/Users/krainboltgreene/.git-hub/config'.

Press <ENTER> to continue, or <CTL>-C to exit.

                        Step #1 - Login ID

The 'git hub' command needs to know your GitHub login id.

Your login is currently set to 'krainboltgreene'.

Do you want to keep this value? [Yn] y

                        Step #2 - API Token

More advanced commands require an authentication token (with the appropriate
scopes added).

To generate a new token, the commands require your GitHub password.

The password will not be stored after the commands finish running.

Enter your GitHub password (for 'krainboltgreene'):

Generating a new authentication token...
Press <ENTER> to continue, or <CTL>-C to exit.
Press <ENTER> to continue, or <CTL>-C to exit.
Do you want to keep this value? [Yn] y

Enter your GitHub password (for 'krainboltgreene'):

# GOTO 10
```

**Comment by Fire-Dragon-DoL** [comment]
Same problem year, can't perform setup, so can't really do a lot :(

**Comment by ingydotnet** [comment]
@krainboltgreene can you run setup with -x, gist the output and put the url in a comment here? This should work:

```
git hub setup -x |& tee output.txt
```

Note that your password will end up in `output.txt` so edit that file and change the occurrences to something like `pa55w0rd` before making it public.

I'll look into seeing if anything changed in the API.

In the meantime, you guys can create a `~/.git-hub/config` file and generate a token on this page: https://github.com/settings/tokens

My file looks like this:

```
[git-hub]
        use-auth = true
        json-lib = json-perl.bash
        login = ingydotnet
        api-token = 1234567890abcdef1234567890abcdef12345678
```

That's all the setup command does is update that text file.

Thanks and sorry for any inconvenience. Hopefully we can resolve this quickly.

**Comment by ingydotnet** [comment]
I just tested setup on the master branch and it worked fine on an ubuntu cloud instance.

Also let me know what OS/environment this is happening in.

**Comment by Fire-Dragon-DoL** [comment]
Sorry forgot to mention, I'm on OS X (yosemite)

**Comment by ingydotnet** [comment]
Hmm. Could be OSX related. @Fire-Dragon-DoL can you do the -x dance for me?

I need to get access to an OSX box. Will try pinging a friend.

**Comment by krainboltgreene** [comment]
I'm afraid it's not terribly helpful :(

https://gist.github.com/krainboltgreene/c34fa69fe4b496df16d6

**Comment by ingydotnet** [comment]
It seems that `git hub token-new -qr 'API Token for git-hub (1433198664)'` is failing and the setup script is not recovering properly. Alas the -x flag doesn't propagate to the subcommand. I should probably patch to make that happen.

I'd like to see the return headers. Try something like this:

```
mv ~/.git-hub /tmp/
git hub setup
for d in ~/.git-hub/cache/*; do [ -d $d ] || continue; echo $d; cat $d/*; done
```

and gist that output.

It appears that you are on OSX and used `make install` to install `git-hub`. Is that correct?

**Comment by krainboltgreene** [comment]
Yes and yes.

Here's the output:

https://gist.github.com/krainboltgreene/c45bfac0381212a958f3

**Comment by Fire-Dragon-DoL** [comment]
@ingydotnet actually I installed with `git clone` and sourced `git-hub/init` (or how was called sorry, the first installation instruction anyway)

**Comment by ingydotnet** [comment]
@krainboltgreene I was aiming for the content of the files in ~/.git-hub/cache/

But it looks like it worked this time. Well all except line https://gist.github.com/krainboltgreene/c45bfac0381212a958f3#file-gistfile1-txt-L44

Did it work? Does your `~/.git-hub/config` file look right??

**Comment by ingydotnet** [comment]
@Fire-Dragon-DoL ok, good to know. I suspect an OSX buglet.

**Comment by krainboltgreene** [comment]
@ingydotnet Newp :/

```
[git-hub]
	login = krainboltgreene
	api-token =
	use-auth = true
	json-lib = json-perl.bash
```

**Comment by ingydotnet** [comment]
Good news. I'm at a conference, borrowed a Mac and  reproduced the issue. Should have it solved shortly!

**Comment by ingydotnet** [comment]
OK, well it gets interesting. At some point in the process after your token is successfully created (and you should be able to see the new tokens at https://github.com/) the `git hub token-get` command is invoked, it succeeds and expects to find a 'token' key in the response.

This key has evidently been deprecated: https://developer.github.com/changes/2014-12-08-removing-authorizations-token/

But for some reason it is not yet deprecated when using curl from Linux.

I'll need to learn the new API and implement it. 

For now, go to https://github.com/settings/tokens and 'Generate new token'. The token will be shown on the screen once. Copy it and insert it in your `api-token` key in the `~/.git-hub/config` file that is generated by `git hub setup`.

Sorry for the inconvenience, but this is the first time I've heard of it. :[

**Comment by ingydotnet** [comment]
Here's a parsed response from an OSX box:

```
/id	19049695
/url	"https://api.github.com/authorizations/19049695"
/app/name	"API Token for git-hub (1434060524) (API)"
/app/url	"https://developer.github.com/v3/oauth_authorizations/"
/app/client_id	"00000000000000000000"
/token	""
/hashed_token	"f9385520226d480157982fa4d9e190e8038a00f57a4712617764e5840fa0667e"
/token_last_eight	"a3f1fef9"
/note	"API Token for git-hub (1434060524)"
/note_url	null
/created_at	"2015-06-11T22:09:02Z"
/updated_at	"2015-06-11T22:09:03Z"
/scopes/0	"user"
/scopes/1	"repo"
/scopes/2	"gist"
/fingerprint	null
```

**Comment by ingydotnet** [comment]
OK Well I was going to follow up with a different response from Linux, but now I get the same empty 'token' value on my Linux box as well.

So now it's not an OSX specific issue. That'a good since I need to return this OSX laptop now.

I'll try to fix this soon.

**Comment by krainboltgreene** [comment]
You are a god. :highfive:

On Thu, Jun 11, 2015 at 5:55 PM, Ingy döt Net <notifications@github.com>
wrote:

> OK Well I was going to follow up with a different response from Linux, but
> now I get the same empty 'token' value on my Linux box as well.
>
> So now it's not an OSX specific issue. That'a good since I need to return
> this OSX laptop now.
>
> I'll try to fix this soon.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ingydotnet/git-hub/issues/163#issuecomment-111299018>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115

**Comment by ingydotnet** [comment]
FYI, I have a fix that is working in the general case. It still is not doing things quite right for entering a wrong password or 2FA code. Still getting it perfect. Also want all the token commands to work right under the new API.

Very soon now.

**Comment by krainboltgreene** [comment]
Awesome. You're amazing!

On Wed, Jun 17, 2015 at 1:33 PM, Ingy döt Net <notifications@github.com>
wrote:

> FYI, I have a fix that is working in the general case. It still is not
> doing things quite right for entering a wrong password or 2FA code. Still
> getting it perfect. Also want all the token commands to work right under
> the new API.
>
> Very soon now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ingydotnet/git-hub/issues/163#issuecomment-112907627>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115

**Comment by ingydotnet** [comment]
I've pushed the fix to master, but without the final touches I wanted to put on
it. I made a new issue #169 for those.

`git hub setup` generates a token correctly now, and that's the most important
thing and worth not waiting on.

Closing this issue. Please test, and reopen if need be.

**Comment by krainboltgreene** [comment]
Awesome thanks :3

On Mon, Jun 29, 2015 at 8:07 AM, Ingy döt Net <notifications@github.com>
wrote:

> I've pushed the fix to master, but without the final touches I wanted to
> put on
> it. I made a new issue #169
> <https://github.com/ingydotnet/git-hub/issues/169> for those.
>
> git hub setup generates a token correctly now, and that's the most
> important
> thing and worth not waiting on.
>
> Closing this issue. Please test, and reopen if need be.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ingydotnet/git-hub/issues/163#issuecomment-116652952>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115
